lib/ansible/modules/network/infinity/infinity.py

```diff
@@ -0,0 +1,650 @@
+            'network_family': network_family,
+            'network_type': network_type,
+            'network_location': network_location}
+        response = self._get_api_call_ansible_handler(
+            method='post',
+            resource_url=resource_url,
+            stat_codes=[200],
+            payload_data=payload_data)
+        return response
+
+
+def main():
+
+    fields = {
+        "server": {
+            "required": True,
+            "type": "str"},
+        "username": {
+            "required": True,
+            "type": "str"},
+        "password": {
+            "required": True,
+            "type": "str"},
+        "network_id": {
+            "required": False},
+        "ip_address": {
+            "required": False,
+            "type": "str"},
+        "network_name": {
+            "required": False,
+            "type": "str"},
+        "network_location": {
+            "required": False,
+            "type": "int",
+            "default": -1},
+        "network_family": {
+            "required": False,
+            "type": "str",
+            "choices": [
+                '4',
+                '6',
+                'dual']},
+        "network_type": {
+            "required": False,
+            "choices": [
+                'lan',
+                'shared_lan',
+                'supernet']},
+        "network_address": {
+            "required": False,
+            "type": "str"},
+        "network_size": {
+            "required": False,
+            "type": "str"},
+        "action": {
+            "required": True,
+            "choices": [
+                'get_network',  # works
+                'reserve_next_available_ip',  # works
+                'release_ip',  # work
+                'delete_network',  # work
+                'reserve_network',  # work
+                'release_network',  # work
+                'add_network',
+                'get_network_id'],  # work
+        },
+    }
+
+    # mutually_exclusive = [ ["network_id", "network_name"] ]
+    # required_together = [ ["attr_name", "attr_value"], ]
+    # module = AnsibleModule(argument_spec=fields, mutually_exclusive=mutually_exclusive, supports_check_mode=True)
+    my_module = AnsibleModule(argument_spec=fields)
+
+    server = my_module.params["server"]
+    username = my_module.params["username"]
+    password = my_module.params["password"]
+    action = my_module.params["action"]
+    network_id = my_module.params["network_id"]
+    released_ip = my_module.params["ip_address"]
+    network_name = my_module.params["network_name"]
+    network_family = my_module.params["network_family"]
+    network_type = my_module.params["network_type"]
+    network_address = my_module.params["network_address"]
+    network_size = my_module.params["network_size"]
+    network_location = my_module.params["network_location"]
+    my_infinity = Infinity(my_module, server, username, password)
+    result2 = ''
```

Comment:
`result2` doesn't seem like a very clear variable name. It's difficult to follow what's going on here